### PR TITLE
refactor: use async fns in minimal example

### PR DESCRIPTION
**Pull request by onbjerg**
Converts the minimal example to showcases that an ExEx is just `async fn(ctx) -> future` and that you can use `async fn`s for both, allowing you to construct ExEx's without implementing `Future` manually

### CODE REVIEW

**Review by DaniPopes — APPROVED** [review]
Can we convert the other examples to use `async fn` instead of `impl Future`? This is much more readable

**Comment by gakonst** [comment]
totally agree with dani, unless there's some non obvious advantage here.

does this also mean we can do 2 exex's that are tokio::join'ed?

**Review by gakonst — APPROVED** [review]

File: examples/exex/async-fn/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    Ok(())
+}
+
+// The initialization logic of the ExEx is also just an async fn!
+async fn exex_init<Node: FullNodeTypes>(
+    ctx: ExExContext<Node>,
+) -> eyre::Result<impl Future<Output = eyre::Result<()>>> {
+    Ok(exex(ctx))
+}
```
Comment (mattsse) [inline examples/exex/async-fn/src/main.rs:31]: nit swap function def order
init
exex

**Review by mattsse — APPROVED** [review]

**Comment by onbjerg** [comment]
I think if we want to convert it all to async fns, I'm going to scrap `async-fn` and just convert `minimal` to an async fn, then in follow ups convert the rest

**Comment by DaniPopes** [comment]
sgtm

File: examples/exex/async-fn/src/main.rs (hunk 1 above)
Comment (onbjerg) [reply examples/exex/async-fn/src/main.rs]: swapped

**Review by shekhirin — APPROVED** [review]
bullish

**Pull request merged**